### PR TITLE
Reproducer for S1905 FP: Floating point arithmetic

### PR DESCRIPTION
**Pull request by gregory-paidis-sonarsource**
Repro for  #9498

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.Test/TestCases/RedundantCast.Fixed.cs [hunk 1]
```diff
@@ -112,4 +112,14 @@ public void Foo(IEnumerable enumerable, IEnumerable<object> genericEnumerable)
...
         public void OfType() => OfType();
     }
+
+    // https://github.com/SonarSource/sonar-dotnet/issues/9498
+    public class Repro_9498
+    {
+        void Sample()
+        {
+            float x = 3.14f;
+            float res = 1 / (float)(x * 2); // Fixed
```
Comment (gregory-paidis-sonarsource) [inline analyzers/tests/SonarAnalyzer.Test/TestCases/RedundantCast.Fixed.cs:122]: We do not fix casts, only `.Cast<T>` and `as T` expressions.

File: analyzers/tests/SonarAnalyzer.Test/TestCases/RedundantCast.cs [hunk 2]
```diff
@@ -115,4 +115,14 @@ public void Foo(IEnumerable enumerable, IEnumerable<object> genericEnumerable)
...
         public void OfType() => OfType();
     }
+
+    // https://github.com/SonarSource/sonar-dotnet/issues/9498
+    public class Repro_9498
+    {
+        void Sample()
+        {
+            float x = 3.14f;
+            float res = 1 / (float)(x * 2); // Noncompliant FP
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.Test/TestCases/RedundantCast.cs:125]: Can you find an example where the precision change is visible?

**Review by Tim-Pohlmann — CHANGES_REQUESTED** [review]
For the reproducer, the result of the calculation does not change. To properly document this, we need to find an example where the change is visible.

As a side note, I do not know how we would implement a fix for this.

File: analyzers/tests/SonarAnalyzer.Test/TestCases/RedundantCast.cs (hunk 2 above)
Comment (gregory-paidis-sonarsource) [reply analyzers/tests/SonarAnalyzer.Test/TestCases/RedundantCast.cs:125]: I don't think it matters, as the solution would probably be what Jared suggested on the post.
Which is to be conservative and raise only in very few circumstances, when the RHS is a variable.

Even more conservative, we could also not raise on floating point arithmetic at all. 

What do you think?

Comment (Tim-Pohlmann) [reply analyzers/tests/SonarAnalyzer.Test/TestCases/RedundantCast.cs:125]: I think we should document a case where this is actually an FP and think about a concrete solution later (or consider it a niche case and ignore it).

**Review by Tim-Pohlmann — CHANGES_REQUESTED** [review]
See the comment above

**Review by Tim-Pohlmann — APPROVED** [review]
LGTM!

**Pull request merged**